lib/stagehand/stagehand.coffee

```diff
@@ -0,0 +1,86 @@
+spawn = require('child_process').spawn
+SdkService = require '../sdk/sdk_service'
+Picker = require '../ui/picker'
+path = require 'path'
+
+# Stagehand support
+class Stagehand
+  @activate: =>
+    return new Promise (resolve, reject) =>
+      cmd  = SdkService.getCommandPath 'pub'
+      args = ['global', 'activate', 'stagehand']
+      errors = ''
+
+      process = spawn cmd, args
+      process.stderr.on 'data', (data) =>
+        errors = errors + data.toString()
+      process.on 'exit', (code) =>
+        if code is 0
+          resolve()
+        else
+          reject(errors)
+
+  @getProjectTemplates: =>
+    return new Promise (resolve, reject) =>
+      cmd  = SdkService.getCommandPath 'pub'
+      args = ['global', 'run', 'stagehand', '--machine']
+      errors = ''
+      json = ''
+
+      process = spawn cmd, args
+      process.stdout.on 'data', (data) =>
+        json = json + data.toString()
+      process.stderr.on 'data', (data) =>
+        errors = errors + data.toString()
+      process.on 'exit', (code) =>
+        if code is 0
+          resolve(JSON.parse(json))
+        else
+          reject(errors)
+
+  @showProjectTemplates: =>
+    picker = new Picker
+    @getProjectTemplates().then (projectTemplates) =>
+      items = projectTemplates.map (pt) =>
+        {item: pt, displayName: pt.label, description: pt.description}
+
+      return picker.selectFrom(items)
+
+  @generate: (projectTemplate) =>
+    picker = new Picker
+    projectPaths  = atom.project.getPaths()
+    generateProject = (projectPath) =>
+      atom.notifications.addInfo "[dart-tools] Generating #{projectTemplate.label}."
+
+      cmd  = SdkService.getCommandPath 'pub'
+      args = ['global', 'run', 'stagehand', projectTemplate.name]
+      errors = ''
+      output = ''
+
+      process = spawn cmd, args,
+        cwd: projectPath
+      process.stdout.on 'data', (data) =>
+        output = output + data.toString()
+      process.stderr.on 'data', (data) =>
+        errors = errors + data.toString()
+      process.on 'exit', (code) =>
+        if code is 0
+          atom.notifications.addInfo "[dart-tools] Generated #{projectTemplate.label}!"
+        else
+          atom.notifications.addError "[dart-tools] Failed to generate #{projectTemplate.label}!",
+            detail: errors or output
+
+    if projectPaths.length is 0
+      atom.notifications.addError "[dart-tools] Cannot run Stagehand without a project directory open."
```

Comment:
I'm not clear what's happening here. Are we requiring there to be a project open to run stagehand? Stagehand will assume that it's running in an empty directory. We probably want to do something similar to what the `Packages > Package Generator > Generate Atom Package` command does; let the user type in a dir name, create that dir, and then run stagehand in the newly created project.